src/raster/rasterio.jl

```diff
@@ -422,14 +422,43 @@ for (T,GT) in _GDALTYPE
             (dataset == C_NULL) && error("Can't read invalid rasterband")
             xbsize, ybsize, zbsize = size(buffer)
             nband = length(bands)
+            bands = Cint.(collect(bands))
```

Comment:
Are you sure this is needed? `ccall` also does conversions as well, it doesn't only take `Vector{Int32}`.

Reply:
You can e.g. change Ptr{Cint} to Ref{Cint} and pointer(bands) to bands, which I think should trigger conversions, but it nevertheless errors for iterable arguments. So I added the explicit conversion. But the cost of this line is marginal so I think it is ok to leave it; furthermore it enables all iterables as arguments e.g. tuples, which is quite a bonus :)

Reply:
Ah yeah I see, good points. Let's leave it in.

Reply:
It does seem like in this case that bands is already a `Vector{Cint}` this could introduce the first allocation in this method, which otherwise mutates the `buffer` argument.
How about something like this to avoid creating a copy when it's not needed?
```
bands isa Vector{Cint} ? bands : Cint.(collect(bands))
```

Reply:
I introduced exactly that, which saves two allocations for all those using the old call signature. How performance critical is this section of the code? In my use case, I call read() a few hundred times at maximum in a non-timecritical context.

Reply:
For the large majority of users I imagine those allocations won't matter. But if we can easily support a way for power users to avoid those allocations I think it's worth it. For instance, now I'm working on a model that loads new geospatial forcing data for every timestep. If the model itself is fast, these allocations can start to matter.